### PR TITLE
Departmental Security Rework Option #1: Department First, Security Second

### PR DESCRIPTION
**Pull request by Iamgoofball**
design doc: https://hackmd.io/@Dsm-BOZJQquXV0Z7ShxJQg/rkwa_htDY

dreamseeker_P3g2gWUAEo

Sprites by Fury McFlurry!

ID trims and HUD icons by Skyrat developers.

## About The Pull Request
Departmental assigned Security is now themed per department.
Medical: Orderly
Service: Bouncer
Supply: Customs Agent
Engineering: Engineering Guard
Science: Science Guard

These jobs all have unique SecHUD icons and ID card trim visuals.

Departmental Security headsets now prioritize the department and not Security.

~~Departmental Security Officers now have more access to their departments so that they can reach emergencies in their department. This is going to be trialled, I might revert this however if it proves to be a bad idea.~~
Scrapped, this was a bad idea.

Service is now available for Departmental Security.

## Why It's Good For The Game

Departmental Security actually staying with their Department is good, actually. It's been scientifically proven that clothing impacts how we operate (http://www.utstat.utoronto.ca/reid/sta2201s/2012/labcoatarticle.pdf) and these uniforms allow Departmental Security Officers to immerse themselves in the department more.

By making the :H chat default to their department, it will also encourage communication with the department more.

~~Allowing Departmental Security to roam their departments more should help them integrate more as well, and respond to emergencies within their department in full.~~
Scrapped, this was a bad idea.

Service was the only department lacking Security presence departmentally, so I have rolled that out as well.
## Changelog

:cl: Iamgoofball, sprites by Fury McFlurry and the Skyrat Development Team
add: Departmental Security has been reworked to have unique outfits and job titles based on their department.
expansion: Service is now available under the Departmental Security system.
qol: Departmental Security headsets now default to their department on :H instead of Security comms.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
and they told me they'd never port deputies upstream

**Comment by LemonInTheDark** [comment]
Does service have department guard spawns?

**Comment by Iamgoofball** [comment]
At the time, no. I just stick them in Security.

**Comment by timothymtorres** [comment]
I don't like the direction of this PR.  Right now, security players are clearly identifiable and I think that shouldn't ever change.  The sprites for this look like a gangbang of assistants trying to lure you into a dark maint tunnel.

**Comment by Iamgoofball** [comment]
None of these look even remotely like assistants.

**Comment by NamelessFairy** [comment]
I feel like the orange shoes clash with the engineering design a lot.
As for the general designs I feel like this hurts identifiability, securities mostly consistent red and black color scheme is easy to identify at a glance, with the exception of the science and engineering ones these blend into the regular staff of those departments for me and I could not identify them as security officers at a glance. The sechud icons also don't help with this, imo they should either maintain the security red with the unique icons, or maintain their colors but share the same default security icon, as to me at least these just look like icons for new departmental jobs.

**Comment by Iamgoofball** [comment]
The goal is to encourage these folks to integrate with their departments more. If they blend in with the staff, that's good.

**Comment by Iamgoofball** [comment]
However, I can totally swap the orange shoes.

**Comment by NamelessFairy** [comment]
> However, I can totally swap the orange shoes.

Yeah I think its just a consistency thing, the design mostly uses black and orangish red, the white and bright orange stick out a lot and unfortunately not in a good way.

**Comment by tralezab** [comment]
imo the only department that has enough internal conflict to justify a department sided guard is medical. the rest simply cut out an important relationship (sec vs crew disputes) by siding sec with a department via their loyalties, instead of the loyalties lying with hos

File: code/modules/jobs/job_types/security_officer.dm [hunk 1]
```diff
@@ -74,55 +76,40 @@ GLOBAL_LIST_EMPTY(security_officer_distribution)
...
+			DEPSEC_DESTINATION = /area/security/checkpoint/supply
+		),
+		SEC_DEPT_ENGINEERING = list(
+			DEPSEC_OUTFIT = /datum/outfit/job/security/engineering,
+			DEPSEC_DESTINATION = /area/security/checkpoint/engineering
+		),
+		SEC_DEPT_MEDICAL = list(
+			DEPSEC_OUTFIT = /datum/outfit/job/security/medical,
+			DEPSEC_DESTINATION = /area/security/checkpoint/medical
+		),
```
Comment (LemonInTheDark) [inline code/modules/jobs/job_types/security_officer.dm:91]: wtf

Comment (Iamgoofball) [reply code/modules/jobs/job_types/security_officer.dm:91]: Replaced the switch loop with this. Is there an issue?

Comment (LemonInTheDark) [reply code/modules/jobs/job_types/security_officer.dm:91]: image

**Comment by LemonInTheDark** [comment]
I want @Mothblocks's input on this, since they at one point had what amounted to a design for departmental sec (didn't write it down in a design doc though, point and laugh)

**Comment by Iamgoofball** [comment]
yeah i went looking after you mentioned it and couldn't find anything so 🤷

**Comment by Iamgoofball** [comment]
aha, found discussions of depsec access, so I rolled back the access changes

**Comment by ATH1909** [comment]
> The goal is to encourage these folks to integrate with their departments more. If they blend in with the staff, that's good.

Until, of course, sec officers end up dead because a non-antag escalates too far against them due to not recognizing them. Sec has a distinctive color scheme so you can recognize the people who are very likely to be on your side at a glance.

**Comment by ATH1909** [comment]
> Security should not be a rolling ball of death swarming antagonists in a giant pile. They should be present all across the station, keeping everyone safe. If this encourages people to stay with their department more, then good.

They, uh, probably SHOULD be a rolling ball of death if they want to be. Plenty of antags can totally dunk on sec officers in 1v1s- cult, for example, is "balanced" around this. Deathballs cover less ground, of course- which is why our sec officers already usually don't seem to stick together (except on, like, Terry with their sec trains).

**Comment by Iamgoofball** [comment]
Cult is also a team gamemode. Trying to 1vX a team is a bad idea.

**Comment by ATH1909** [comment]
Like, I do agree that helping out your assigned department is a constructive way to pass the time as a departmental sec officer (although I don't believe that we should force that on people), but if someone shouts "HERETIC MEDBAY" over comms, your priority should be dropping everything and getting over there to reinforce, not pushing crates.

**Comment by Colovorat** [comment]
> The goal is to encourage these folks to integrate with their departments more. If they blend in with the staff, that's good.

Departamental sec is sec that got assigned to guard departament, not departament job that got assigned to a sec. They are security first, then they are departament guards.
Also this sprites look like literally greytider stole sec helmet and armor. Current security design is very good, it's really easy to distinguish sec officer and tider that stole sec gear

**Comment by ATH1909** [comment]
> Cult is also a team gamemode. Trying to 1vX a team is a bad idea. Skill issue, etc.

so is trying to 1v1 a guy with an ebow

or a changeling who's made a stun prod

or a heretic with a funny hand, a way to deal that last bit of stamina damage, and an escape spell

**Comment by LukasBeedellCodestuff** [comment]
I've got a lot of security hours under my belt and I've got some input here:

I like giving security more access in their departments that's super cool and we should do that anyway but why do we need to make them guards? The sprites look less like a security officer and like someone went to the clothes vendor and got clothes. The sechud sprites also look kinda dumb and cannot be instantly taken as sec, might have to double take on a few of those to think "is this virology or what" for example.

You're taking away the organic security look of sec for hopes that they integrate with their department and don't run off to kill valids. If you're a shitter player having clothes won't help you be nicer and that's something we need to look at. It also doesn't combat the problem of people running off at all it just gives them roundstart vendor clothes which is pointless. The us vs them dynamic will not be dismantled by giving them these pretty clothes.

Tldr; Bloat pr, 0 positives and makes it harder to see who's sec and who's not. Put clothes in vendor, leave sec untouched.

This is just a bloat PR.

**Comment by UnokiAs** [comment]
Maybe making the hud Icons red would make it less confusing for security

**Comment by LukasBeedellCodestuff** [comment]
> Maybe making the hud Icons red would make it less confusing for security, And the medbay symbol a cross

Then they would just look super ugly.

**Comment by UnokiAs** [comment]
>Then they would just look super ugly.

Not the true sec color(Made it in paint imo), But I think making red still look better for information

141080786-82cb40c7-6468-47f6-af4d-da40051297d4

**Comment by Jackraxxus** [comment]
I have four comments about this PR.
Firstly, I don't see jumpskirt versions of any of the proposed uniforms in Icon Diff Bot's little readout. I might just be blind though. I need my wound bonus vs bare skin for targeting legs on seccies pls.
Secondly, I like sec patrolling around. I tried sitting in my assigned department once and it was boring. The only department that really benefits from a dedicated officer is medical and only if that officer can stop bomb threats. Like seriously I once collected 3 minibombs, a bag of X4, 2 ttvs, and a syndicate bomb, in the same round as medsec on MRP. 
Thirdly, I like that it makes the loot piñata that is engisec more identifiable. Don't have to examine seccies to find out which one has the free toolbelt and RCD.
Finally, this is just making sec blue with extra steps just please bring bluesec back it will solve all of your issues trust me.
Maybe this (And bluesec) could be fun as like a roundstart station trait? That feature doesn't have nearly enough bloat yet.

**Comment by Googles-Hands** [comment]
> The sprites look less like a security officer and like someone went to the clothes vendor and got clothes. The sechud sprites also look kinda dumb and cannot be instantly taken as sec, might have to double take on a few of those to think "is this virology or what" for example.

To add to that, this will heavily mess up the HoS and HoP when they want to redistribute officers to different departments since they all look so different.

Overall a PR that doesn't fit the style of security at all.

**Comment by InsaneRed** [comment]
every "departmental guard" just looks like a tider from said department took sec armor and put it on

**Comment by necromanceranne** [comment]
Man I was literally thinking about this the other day. And while I'd go even further with it and borrow your original idea for guards this is a solid first step to involving security more in their assigned department and less of a weird mallcop paramilitary force.

**Comment by Pickle-Coding** [comment]
Why does it have the QoL tag?

**Comment by InsaneRed** [comment]
> 
> 
> Man I was literally thinking about this the other day. And while I'd go even further with it and borrow your original idea for guards this is a solid first step to involving security more in their assigned department and less of a weird mallcop paramilitary force that look too out of place being anywhere at all.

how does this make them fit in while all of them look like non sec that just armed themselves with sec gear?

**Comment by jlsnow301** [comment]
>Engineering: Engineering Guard

Really missing the chance for OSHA roleplay here
```Safety Inspector```

**Comment by Iamgoofball** [comment]
lmfao

**Comment by Mothblocks** [comment]
> I want @Mothblocks's input on this, since they at one point had what amounted to a design for departmental sec I think (didn't write it down in a design doc though, point and laugh (maybe at me we'll see when they wake up))

I didn't really have a design, I was just testing how matching them together would work with @ninjanomnom (answer: it worked okay, but our maps aren't really suited for it ATM)

**Comment by Mothblocks** [comment]
That being said I'll echo that I don't really like the art

File: code/__DEFINES/preferences.dm [hunk 2]
```diff
@@ -64,6 +64,7 @@
 #define SEC_DEPT_MEDICAL "Medical"
 #define SEC_DEPT_SCIENCE "Science"
 #define SEC_DEPT_SUPPLY "Supply"
+#define SEC_DEPT_SERVICE "Service"
```
Comment (Mothblocks) [inline code/__DEFINES/preferences.dm:67]:
```suggestion
#define SEC_DEPT_SERVICE "Service"
#define SEC_DEPT_SUPPLY "Supply"
```

File: code/_globalvars/lists/flavor_misc.dm [hunk 3]
```diff
@@ -185,6 +185,7 @@ GLOBAL_LIST_INIT(security_depts_prefs, sort_list(list(
 	SEC_DEPT_MEDICAL,
 	SEC_DEPT_SCIENCE,
 	SEC_DEPT_SUPPLY,
+	SEC_DEPT_SERVICE,
```
Comment (Mothblocks) [inline code/_globalvars/lists/flavor_misc.dm:188]:
```suggestion
	SEC_DEPT_SERVICE,
	SEC_DEPT_SUPPLY,
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Ok and especially with the names this is absolutely going to be really confusing on telling who's security and who's not, really don't think I like this

**Comment by Mothblocks** [comment]
wait i actually said one of those last comments before, just in the context of access

image
image

**Comment by Iamgoofball** [comment]
>That being said I'll echo that I don't really like the art in terms of consistent recognizing of security.

as I have said on discord, if you want adjustments to the sprites made, you will need to foot the bill for paying the spriters to do work

however, i will be adjusting the HUD and ID icons

**Comment by Iamgoofball** [comment]
in the absolute worst case scenario, one can just shift click someone to see they have a red square on their ID and their brain can go "oh security person"

**Comment by Iamgoofball** [comment]
additionally, regarding the access, Timberpoes stated they didn't want secoffs to be able to have supreme access in departments just by being departmentals, so I've removed the access changes from this PR.

**Comment by Mothblocks** [comment]
yeah i was just referring to the last time i've spoken about dept officers, i don't really have as much of a foothold in them as some people think

**Comment by Mothblocks** [comment]
also a lot of weird "need to"s when the pr is not a requirement to have

**Comment by o0cyann0o** [comment]
This is deranged. Security is it's own department, it should feel, act, and look like it. This also muddies who secoffs should listen to, their actual head of fuckin staff or the cmo/rd/qm or whatever

**Comment by o0cyann0o** [comment]
having service seccies is based though thats actually a really good idea

**Comment by Iamgoofball** [comment]
>This also muddies who secoffs should listen to, their actual head of fuckin staff or the cmo/rd/qm or whatever

They're supposed to listen to both. It's been like this for year
chrome_VflMJjYVEA
s.

**Comment by Farquaar** [comment]
I'm going to be blunt. These sprites are completely unsuitable. They look like some assistant visited the autodrobe after stealing a security helmet. As many people have been saying already, you need to be able to identify security at a glance. That's the purpose of a uniform in the first place.

Additionally, the idea that departmental security should be relegated to their department doesn't seem well-founded. Sure, an engineering guard will spend some time in engineering at the beginning of the shift, but when a radio call goes out to stop a murder in progress on the other side of the station, all seccies, regardless of department, will come running. Same when there's a manhunt, or a cult base, or a traitor on a shooting spree. Departmental security officers are security officers first and foremost. The departmental designation is just to give them something to do on their downtime.

**Comment by jacklecroy** [comment]
I think this is a cool idea if the sprites are improved.

**Comment by Iamgoofball** [comment]
>Departmental security officers are security officers first and foremost. The departmental designation is just to give them something to do on their downtime.

Yes.

During this downtime, this PR makes it go from "Security is hanging around my department" to "a member of my department is hanging around my department".

That is the impact clothing has.

**Comment by Iamgoofball** [comment]
>I think this is a cool idea if the sprites are improved.

The actual clothing themselves is fine, I'll be fixing the cards/hud icons shortly.

**Comment by jacklecroy** [comment]
> The actual clothing themselves is fine, I'll be fixing the cards/hud icons shortly.

To have the same icon but different colors?

For the clothing, I think they should look more similar. Like current security but with different color marks on them. The bouncer looks like an assistant. The cargo one looks like a cargo tech. The medbay one looks like an assistant. I get they are wearing helmets, but people wear helmets all the time already. Especially cargo techs. I like the black mesa style science outfit.

**Comment by Farquaar** [comment]
>
> During this downtime, this PR makes it go from "Security is hanging around my department" to "a member of my department is hanging around my department".

This is precisely the problem.

**Comment by Iamgoofball** [comment]
I'm sorry that you don't understand the problem.

**Comment by Farquaar** [comment]
> 
> 
> I'm sorry that you don't understand the problem.

What exactly is the problem with security being identifiable at a glance?

**Comment by anconfuzedrock** [comment]
I could say things that were already said but what I will say is that most people in departments hardly ever stay in them, and they have more reason to be there, what makes you think sec guards with nothing but a cam console will stay? And security is even LESS intrinsically motivated to work in the department they get for another reason- they aren't expecting to get it. There's only one slot for each department guard and you can only pick one as your main preference, so either you get it, which you weren't even really expecting, or you don't get it, so now you're in a department you don't want. Compare this to most departments, where people pick the role that has several slots and expects to get it, and if they don't get it then they can pick another specific preference/latejoin. 

Lastly this tries to create a problem that is also made worse by adding yet another department guard. and because getting a department is generally a good thing, you can expect most officers will have a preference for a department guard selected. So unless there are at least six security officers (lol) all are going to be assigned to departments which you don't want them to leave. Good luck if you're the warden or HOS trying to coordinate officers for ANYTHING. best off just doing all the work yourself HOS, you're the only officer who isn't intended to be glued to a seat.

**Comment by RaveRadbury** [comment]
There's a hydroponics arm band on meta I think, always assumed that was a weird service sec hint.

**Comment by Iamgoofball** [comment]
@anconfuzedrock 
>There's only one slot for each department guard
This is wrong. That's not how departmental security assignment works. Look at the code for details of the algorithm breakdown that @Mothblocks wrote up regarding departmental security assignment.

**Comment by optimumtact** [comment]
once again a major change with no hackmd, I'm not sure the process is understood goof, it's a design document first that is sighted by maintainers and headcoders and signed off on, before you run about trying to make the change.

**Comment by Iamgoofball** [comment]
ah yes so how exactly do you propose i get that design document in front of maintainers oranges

**Comment by Iamgoofball** [comment]
this isn't even a major change, oranges

it's fucking clothing and a tweak to a few radio headsets

**Comment by optimumtact** [comment]
Goof I don't give a flying fuck if you think this is the goddamn ship of thesus.

Write up a design document and ping me with it, print it out and fucking snail mail it if you want, generate some smoke signals with your thoughts and feelings.

I literally could not care how you want to do it, Just stop making wholesale changes to the game without checking first, it's not 2015 anymore.

**Comment by maxymax13** [comment]
goof tier pr and sprites

**Comment by Googles-Hands** [comment]
So will this remain testmerged until Goof can get that design doc (hopefully never) or will it be completely taken out soon?

**Comment by Iamgoofball** [comment]
https://hackmd.io/@Dsm-BOZJQquXV0Z7ShxJQg/rkwa_htDY

design doc made

**Comment by InsaneRed** [comment]
image
can you expand on how you think this is a good idea? you are not only splitting off people with sec acess into basically other departments as you yourself said that they are diffrent from sec, and making them be able to join revs/cult/families this basically gives off the problem that sec is already cut in half and half of them are antags how is that balanced to sec players?

**Comment by RuneQuester** [comment]
image


Thank you goof

**Comment by Mothblocks** [comment]
From what I've been told, that bug should already be fixed @RuneQuester

**Comment by Iamgoofball** [comment]
@InsaneRed 
>can you expand on how you think this is a good idea? you are not only splitting off people with sec acess into basically other departments as you yourself said that they are diffrent from sec, and making them be able to join revs/cult/families this basically gives off the problem that sec is already cut in half and half of them are antags how is that balanced to sec players?

this isn't in this PR, you have the wrong PR for this

**Comment by Zytolg** [comment]
I'm not entirely convinced that /tg/ needs this. Our security works well as-is, and guards assigned to a department generally carry out their duties. In the case that security players would be thugs, which is something that happens unsurprisingly often, disgruntled players and even departments generally fight back. Your only argument for this is that clothes affect psychology, but you could just add the drip in to the preexisting secposts in a much less impactful PR. Decentralizing sec would also lean the advantage of conversion gamemodes towards the "bad guys". 

Breaking things down a bit more. 
Bouncer: This is unneeded. The Barkeep gets a shotgun and the chef gets CQC to combat would-be griefers. The Clown deserves whatever is coming to them, and botany is an arms factory (don't even begin to pretend it isn't).
Cargo: Cargo sec is already pushing it. Cargo doesn't need a security officer when half the time, they're more well armed than the department that's _supposed_ to be protecting them.

Now I can actually understand why we have a science sec post considering that this is a research station and there are those who would target Nanotrasens top secret research. Engineering is in the same boat. The Supermatter is a highly experimental engine after all. But in the current implementation of Security, those posts and positions are already covered.

So I poise the question: Just what exactly are you seeking to change here.

**Comment by optimumtact** [comment]
maybe you should read the PR before asking dumbass questions

**Comment by Farquaar** [comment]
>Problems with Departmental Security as it stands right now:
>
>1. Departmental Security just determines what gamer gear a security officer gets roundstart via proximity, be it medical tools, engineering tools, or whatever the fuck you can loot from science roundstart, honestly there’s not much

If a player chooses not to do their job as department security, they should disable department security in their preferences.

>2. Departmental Security doesn’t actually interact with the department outside of who’s at the front of the conga line if the HoS isn’t available.

Again, if a player chooses not to do their job as department security, they should disable department security in their preferences.

>3. Departments hate their Departmental Security Officers because those Departmental Security Officers are either taking shit roundstart and disappearing with it, or buzzing the rest of Security in the front door to ruin everyone’s day, or both at the exact same time.

Press X to doubt. In my experience people like having a security officer in their department so long as he's not a dick.

>4. Departmental Security has absolutely zero loyalty or reason to integrate with the department they’re assigned to, resulting in incredibly baffling situations such as the Medical Security Officer gunning down medical doctors for providing care to patients they didn’t want revived, or Engineering Security Officers stealing insulated gloves so that they don’t have to deal with

I don't even know where to begin with this. If you see somebody breaking server rules by "gunning down medical doctors" and stealing vital equipment for no reason, ahelp it or mutiny. This has nothing to do with "departmental loyalty".

>5. Any attempts to stop 4 results in immediate round removal because Security has all the guns and firepower immediately roundstart that the crew does not have the time to make.

What server are you playing on where security is gunning down half the station at the start of the shift for no raisin?

>All five of these problems lead to Departmental Security basically being an excuse for free, roundstart

...what?

**Comment by Colovorat** [comment]
> ah yes so how exactly do you propose i get that design document in front of maintainers oranges
> 
> because so far every time I get a conversation going with maintainers, certain members of your team shut the discussion down completely and refuse to let me to do and then report me for off topic posting

Maybe if you don't come up with weird ideas over and over you will not be ignored, just a suggestion

**Comment by NanoCats** [comment]
I like it :)


Why the fuck is engieguard's symbol a peace sign though?

**Comment by necromanceranne** [comment]
> I like it :)
> 
> Why the fuck is engieguard's symbol a peace sign though?

Peace Walker reference maybe?

**Comment by Razharas** [comment]
They should either definitely look like security or definitely look like they are a doc/cargonian
Currently they look like they dont belong anywhere but maint tunnels

**Comment by Razharas** [comment]
On a more specific note, the problem with dept sec is exactly singular - they never stay and guard their department
A solution to this may be something akin to "guarding duty", make it so one of the department dudes randomly gets a bright red armband of their shoulder and a pepper spray/something else defensive like this, maybe sec radio key too, so that there is always at least one "defender" of a department, and make all of these things easily transferable in case someone else would rather be in charge of that

**Comment by wesoda25** [comment]
This seems like a step in the wrong direction to me. I feel as if we're just going to see more assistants valid hunting because of this and the line of who can deal with them will blur. Seems ultimately more frustrating than the current experience with security imo.

**Comment by Kapu1178** [comment]
I hate Goof Sec on Skyrat and I wouldn't wish it on any other server. It's basically just an excuse to tell sec to go screw, doesn't improve gameplay what so ever.

**Comment by Googles-Hands** [comment]
>A solution to this may be something akin to "guarding duty", make it so one of the department dudes randomly gets a bright red armband of their shoulder and a pepper spray/something else defensive like this, maybe sec radio key too, so that there is always at least one "defender" of a department, and make all of these things easily transferable in case someone else would rather be in charge of that

This would go against what security on tg is, it is it's own independent department and while you may be assigned to guard another department, you are still first and foremost a member of security, not some department worker that got combat training and weapons.

**Comment by Razharas** [comment]
I didnt mean this would be instead of dept sec
I meant that this would be a good thing besides a dept sec, just usual doc/cargonian/etc spawning with a bright red armband and a pepper spray

**Comment by ShizCalev** [comment]
thanks i hate it

**Comment by EricZilla** [comment]
There are a lot of good ideas here, and I would really like to see like, DepSec Rework Lite as an option, wherein you get the cool ID stuff, get your custom hat and armor vest, but have normal sec gear otherwise.

I really like all the sprites btw, barney science is such a god tier reference too. I think having a policy infobit on joining as departmental security would also be a good step in the right direction. "Your mission is to protect your department personnel and workspace above all else. You report directly to both your assigned department head AND the head of security, and are expected to let minor threats outside your assigned area be handled by the appropriate people"

Or like, something worded you know, well.

**Comment by Moltovz** [comment]
I love the ID sprites- Hate everything else

**Comment by drIcepop** [comment]
> I love the ID sprites- Hate everything else

hate the ID sprites, hate everything else

**Comment by HansRobertCoooding** [comment]
I seriously hate the HUD icons, They look like someone made an AI randomize HUD icons.

**Comment by RuneQuester** [comment]
image
Goof. Plasma-men don't spawn with envirosuits. How did you fucking forget this.

**Comment by zxaber** [comment]
Not a big fan of the outfits. Department first, security second, but they *are* still security and should be recognizable as such at a glance.

Maybe keep the normal Sec jumpsuits but give them a department-themed jacket or body armor? Security red underneath, but showing department colors? (This is basically what department armbands did but I understand if the argument is that the armbands were hard to notice).

If we don't want these dept sec to act as sec at all, we should just use rework option#3.

**Comment by Zytolg** [comment]
Goofball, fix your fucking orderlies barging into chemistry and batoning the chemists so that they can make meth.

**Comment by CMDR-Gungnir** [comment]
> Not a big fan of the outfits. Department first, security second, but they _are_ still security and should be recognizable as such at a glance.
> 
> Maybe keep the normal Sec jumpsuits but give them a department-themed jacket or body armor? Security red underneath, but showing department colors? (This is basically what department armbands did but I understand if the argument is that the armbands were hard to notice).
> 
> If we don't want these dept sec to act as sec at all, we should just use rework option#3.

They'll look like Security when you get used to their appearance. Many Seccies already opted to wear the formal uniform or the Blue Shirt and Tie anyway, so what difference does it make? You still learned to associate it with Security, right? There's also the handy tool of "shift click and read the ID" when you see something you don't recognise.

> Goofball, fix your fucking orderlies barging into chemistry and batoning the chemists so that they can make meth.

To be fair, this could and would happen with DepSec too.

**Comment by LemonInTheDark** [comment]
Regardless of if this is a good idea or not, what the heck is he supposed to do about people being assholes

**Comment by Colovorat** [comment]
> Regardless of if this is a good idea or not, what the heck is he supposed to do about people being assholes

Exactly
Nothing

**Comment by Iamgoofball** [comment]
attempting to baton.a department member calls gib

**Comment by necromanceranne** [comment]
Radical suggestion but if you want these guys to be departmental focus, maybe actually keep them locked to specifically main security and their departmental maint only. The biggest reason why most departmental sec leave their department is because they're probably just roaming around in maint as normal, but using department access to get all the goodies inside to piss off again. 

If they're _effectively_ just department members with weapons, that might encourage them to stick around and do department work when they're not doing security work.

**Comment by Mothblocks** [comment]
That would be a huge step in a negative direction that I don't want to see take place, I still fundamentally disagree that departmental officers not sticking to their department is a bad thing.

**Comment by Iamgoofball** [comment]
@Mothblocks If you just purely care about officers sticking together, spawn them in the brig and remove departmental security. Departmental Security is actively counter-productive to whatever you want here.

**Comment by Iamgoofball** [comment]
basically mothblocks, if you want a buddy cop system, *code a buddy cop system* that isn't just a thinly veiled excuse for security to greytide departments roundstart for gamer gear

**Comment by Mothblocks** [comment]
I've been leaving this test merge up to see how people interact with it once the novelty wears out to see if it changes anything, then if the maintainers still think something needs to be changed then I'm happy to split security into smaller dedicated teams without accesses.

**Comment by Iamgoofball** [comment]
>if the maintainers still think something needs to be changed then I'm happy to split security into smaller dedicated teams without accesses.

That isn't at all what the discussion in #maintainerbus said.

**Comment by newfren** [comment]
The security guard alternate names are out of alphabetical order in the id imprinter and the access management software.

**Comment by Orchian** [comment]
Good stuff

**Comment by LukasBeedellCodestuff** [comment]
Please fix crew sensor if this is going to stay merged. Security officers are showing as invidividual and cap coloured on the sensors instead of as sec.

**Comment by Coastfront** [comment]
1. Why does service even really n e e d a guard? The chef has cqc, The bartender has a gun that they almost always replace its ammo with lethals if someone is annoyinng enough, and botany gets aids plants and hatchets. The rest of the members of service either literally never go in their departments area for more than 2 seconds, or have a nicher form of weapon (beartraps.) just about everyone in service who has a room that they'd want protected already spawn with/can easily make a weapon with which they can defend said rooms. 

2. As others have said and i don't mean this in a destructive way, but the id sechud sprites look, not good. Most job sechud icons are meant to quickly be matched to their department, even if you ignored the colorscheme. Engineers having a wrench, medical having a medical cross, chef having a fork and knife, etc. The departmental sec icons, except for the peace symbol, dont really match/make sense. If someone has a sechud, and walks near a departmental seccie, They should be able to tell at a glance which department they're supposed to be guarding if they swapped their color coded clothes. What the actual fuck, is 3 dots and an exclamation mark supposed to mean??? or a circle with a pie slice taken out of it? 

This last one is less a complaint on the pr's contents specifically, and more the direction the pr goes. Departmental security officers, due to the nature of the role, will almost never if ever decide to stay in their department and actuall guard it. Since even if you change how it looks, Even if you re-prioritise their radio, The job they picked said security officer. Not medical officer. A more effective way to encourage departmental sec to actually try to defend their department would probably have to be just detaching the sec outposts from sec officers, and making a new job entirly who has less sec acess and more department acess. Again this last bit is less a criticism of the contents of this pr specifically and more a criticism of how this pr is adressing the issue it is, and the direction this pr would go.

**Comment by RuneQuester** [comment]
Goof. Have you fixed plasma-men not starting suits. Please answer. You ignored the screenshot I posted of this

**Comment by drIcepop** [comment]
This PR has had a negative reception, yet it's still going to get merged because "It's what the codebase needs"

**Comment by Mothblocks** [comment]
Who are you quoting

**Comment by optimumtact** [comment]
reddit moment

**Comment by Bawhoppen** [comment]
This is secretly a continuation of the security sprite wars
A clear violation of the Steelpoint Treaty of 2015

**Comment by realthomas69** [comment]
when i set my preferred assigned department to "none" it seems to just ignore that and assign me to a department anyway, so i cant really opt out of this, pretty lame.

**Comment by Mothblocks** [comment]
It means no preference, not preferring nothing.

**Comment by Googles-Hands** [comment]
Since this wasn't shut down long ago, does this mean the Headcoders agree with the, albeit provocative, statement "Department First, Security Second" which this PR is titled with and that this will be the new direction security development will be taking.

Furthermore, do the "Problems" presented in the design document really get fixed by these solutions, how does giving Security Officers new clothes fix that people elect to guard Engineering for insulated gloves?

How does it make them more "loyal" to the department they are guarding? Goof's example of "get an assistant job changed and stand in medbay" does not apply because a player of a previous DIFFERENT job changes to a NEW job entirely. We still expect Security Officers to uphold order, their task remains the same. Dressing up an Officer like Barney does not mean that they will sit in their Department Post all round and constantly patrol it. They will still patrol the entirety of the station because thats what being proactive means.

Also as others have said, the service Officer is not needed at all and is just bloat. The Bartender has a shotgun, the cook has CQC and the janitors are not in the "service" area as they move around the station constantly and will either slip or just call for help on the radio while in an department/in the main halls.
The only job that would theoretically "need" protection is Botany, and there already exists the "Hydroponics Guard" armband if you really want to post an officer in the mess hall/botany area all round.

**Comment by optimumtact** [comment]
>Since this wasn't shut down long ago, does this mean the Headcoders agree with the, albeit provocative, statement "Department First, Security Second" which this PR is titled with and that this will be the new direction security development will be taking

Agreement is given, not earned through our inaction, headcoders don't have a position on this, when there is a specific design decision for security, there will be a related design document on the tgstation hackmd.

**Comment by GreenTea98** [comment]
Haha holy shit this is stupid

**Comment by optimumtact** [comment]
this is being put to a vote internally among the maintainers as no clear consensus came out in discussion,.

**Comment by Iamgoofball** [comment]
For the record, this also hasn't been testmerged for about 12 days so that Modsuits could be testmerged, since we change the same icon DMIs.

**Comment by optimumtact** [comment]
Dept sec vote has concluded, the outcome of which is that we will not go ahead with dept sec changes.

I also asked maintainers to determine which of the proposed solutions they liked the most (if we went ahead with the change)
Of the proposed options in the design document https://hackmd.io/@Dsm-BOZJQquXV0Z7ShxJQg/rkwa_htDY

the vast majority of maintainers would have preferred solution 2, with a small minority interested in solution

**Comment by optimumtact** [comment]
handsclaspedglasses

**Comment by LemonInTheDark** [comment]
> with a small minority interested in solution

**Comment by Iamgoofball** [comment]
i opened option #2 literally day of and got told no by mothblocks @optimumtact 
https://github.com/tgstation/tgstation/pull/62758

**Comment by optimumtact** [comment]
I don't see how that's relevant to anything I said.

**Comment by Iamgoofball** [comment]
re: ``the vast majority of maintainers would have preferred solution 2,``
the implication on that PR closure was that maintainers did not prefer solution 2, and therefore we didn't do any testing on it

**Comment by Mothblocks** [comment]
*Would have* preferred solution 2 *if* we wanted any change at all
